### ISSUE TITLE
Refactor the Config class

### ISSUE THREAD

**Issue by davidrpugh** [EconomicSL/housing-model#39]
@adrian-carro I would recommend using a config library rather than re-inventing the wheel with the current Config class.  I have used [HOCON](https://github.com/typesafehub/config) in my previous Java/Scala work and it seems solid.  

Once I finish the reorg PR, I will open a PR that swaps out the current Config class for one that uses the HOCON configuration library instead.

**Comment by adrian-carro** [comment]
Oh, that library looks great! The main reason I didn't go for JSON was the absence of comments. I didn't know about this HOCON which, I assume, is a JSON superset with comments and other stuff, right?

**Comment by davidrpugh** [comment]
@adrian-carro Correct.  Here is a link to the Akka [reference config](https://github.com/akka/akka/blob/master/akka-actor/src/main/resources/reference.conf) for an example of what the config file would look like.

**Comment by rht** [comment]
In my python port of housing-model, I simplified the config by putting everything into a python class (https://github.com/rht/housing-model/blob/master/src/main/java/housing/Config.py). This way, no more parsing of a custom config file format is needed.
